### PR TITLE
Added staging command

### PR DESCRIPTION
**Pull request by ghost**
Hi,
  Really excited about this tool, it seems like it will fix exactly the kind of work flow I'm looking for.  One thing I needed though was the ability to push to staging in addition to pushing to production.  Because we're just talking about pushing to another branch and not deleting the remote branch, it was pretty easy to implement.  Let me know what you think.
  Best,
Rob

### CODE REVIEW

**Comment by pcreux** [comment]
Hey!

I'm very happy to know that you like it! :) Thanks for your pull request.

In our workflow, master is not the production branch but the development one. All the "reviewed" branches are merged onto master using gitmine. Then we merge master to staging (and deploy), then merge staging to production (and deploy).
We do that using mothership (https://github.com/versapay/mothership/).

We decided to use this workflow because we often break tests when we merge things to master and we want the deployment to staging to be the exact same as the one to production.

Do you need to add this new command if you use this workflow instead? 

Cheers!

**Comment by notch8-old** [comment]
The problem with that work flow is that you can't pull individual features out when it is time to do a deployment if they do not work on staging (or more commonly if the client isn't happy with what they asked for).  We frequently will have a batch of tickets (call them [1, 2, 3, 4, 5, 6]) representing user stories.  Our user stories are very atomic and complete.  All go to staging through out the week, but only 1, 4, 6 get approved to go to production.  Often 1 and 6 are high priority and need to go out that week.  So those features get merged in to a fresh staging branch and pushed up for a quick run through and then merged to master while the others go back into active dev.  That simply isn't possible with your approach unless I'm missing something. 

Ultimately I'd like to add a little more flexibility to the tool, but keep the reviewed / stage concepts around in some form.  There may be better names for them.  I think if you have the ability to merge into a configurable branch (defaults to staging) OR merge into a configurable branch (defaults to master) and delete a initial branch, coupled with the ability to set the status you go to (already have that) gitmine can accommodate a wide variety of workflows.

Bottom line, no, but unless I'm missing something, I don't think this approach eliminates my need to be able to merge w/o deleting the branch.

Rob

**Comment by pcreux** [comment]
We don't have the same needs: we don't deploy to staging to review a
feature, we do it locally.

Also I wasn't asking you to switch to our workflow. I was wondering if
you weren't using master as your production branch since this is the
default in  https://github.com/nvie/gitflow.

What about getting to configurable merge commands in .gitmine.yml:

``` yaml
merge_commands:
  staging:
    merge_to: staging
    delete: false
    update_status: 23
    aliases:
      - stag
      - for_qa
```

Cheers,

**Comment by notch8-old** [comment]
Hi,
  Sorry, guess I miss-understood.  We definitely use master for a our production branch, as most of our customers are deployed to Heroku at this point and that makes life easier.  We do testing locally, but merge to a staging branch of some kind in order to do CI and customer approval.

  The YAML approach would be totally customizable, with out adding that much complexity.  I like that.  I'd think we'd definitely want to support writing a more complete example config file for people if we go that direction.

=-)
Rob
